pkg/virtctl/credentials/credentials.go

```diff
@@ -233,6 +300,54 @@ func createPatchToAddKeyToSecret(keyName string, key []byte) []byte {
 	return payload
 }
 
+func removeKeyFromSecret(ctx context.Context, cli kubecli.KubevirtClient, secretNamespace string, secretName string, key string) error {
+	// Looping, because Update API call can fail with conflict
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+		}
+
+		secret, err := cli.CoreV1().Secrets(secretNamespace).Get(ctx, secretName, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			// Secret does not exist, nothing to do
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+
+		for fileName, data := range secret.Data {
+			updatedData := removeSshKeyFromBytes(key, data)
+			if len(updatedData) == 0 {
+				delete(secret.Data, fileName)
+			} else {
+				secret.Data[fileName] = updatedData
+			}
+		}
+
+		_, err = cli.CoreV1().Secrets(secretNamespace).Update(ctx, secret, metav1.UpdateOptions{})
+		if errors.IsNotFound(err) {
```

Comment:
nit: you already checked the secret existence above

Reply:
The secret could have been removed in the time between the calls. If that happens, I don't want the command the fail with an error.